packages/cells/style/widget.css

```
@@ -71,8 +71,9 @@
 
 .jp-CodeCell.jp-mod-outputsScrolled .jp-Cell-outputArea {
   overflow-y: auto;
-  max-height: 24em;
+  height: 24em;
```

```suggestion
  max-height: 24em;
```

When users enable scrolling for outputs for all cells in the notebook they would end up with some outputs taking up more space than before.

There is a neat CSS trick allowing to retain `max-height` here, which relies on the resize handler adding `style="height: {userSetValue}px"` once user starts resizing, so we can target this state with a separate CSS selector:

```css
.jp-CodeCell.jp-mod-outputsScrolled .jp-Cell-outputArea[style*=height] {
    max-height: unset;
}
```

This should be safe to rely on as this behaviour is standardised in the spec:

> When an element is resized by the user, the user agent sets the [width](https://drafts.csswg.org/css-sizing-3/#propdef-width) and [height](https://drafts.csswg.org/css-sizing-3/#propdef-height) properties to px unit length values of the size indicated by the user, in the element’s [style attribute](https://www.w3.org/TR/css-style-attr/#style-attribute) DOM, replacing existing property declaration(s), if any, without !important, if any.

(https://drafts.csswg.org/css-ui/#resize)

---

Oh, that's really nifty - thanks for the advice!